docs: Fix incorrect nested LLM reranker configuration structure #3803

priyanshujaiz opened the pull request:
## Description

This PR fixes a discrepancy between the documentation and the actual codebase regarding the LLMReranker configuration. The documentation previously illustrated a nested "llm" dictionary structure, whereas the LLMRerankerConfig class and initialization logic require a flat dictionary structure.

This update ensures the configuration examples align with mem0/configs/rerankers/llm.py and prevent runtime errors for users following the documentation.

Problem
The documentation previously provided configuration examples with the following nested structure:

"config": {
    "llm": {
        "provider": "openai",
        "config": {
            "model": "gpt-4",
            "api_key": "..."
        }
    }
}
However, the codebase implementation does not support this nesting:

LLMReranker.__init__ directly unpacks the configuration via LLMRerankerConfig(**config).

There is no logic to handle or unpack a nested "llm" key.

The parameter names in the documentation (e.g., custom_prompt, top_n) did not strictly match the Pydantic model definitions.

SOLUTION
I have updated all relevant documentation examples to use the correct flat structure and accurate field names, aligning them with mem0/configs/rerankers/llm.py.

Corrected Structure:

"config": {
    "provider": "openai",
    "model": "gpt-4",
    "api_key": "...",
    "temperature": 0.0,
    "top_k": 10
}
Changes Made
1. docs/components/rerankers/models/llm_reranker.mdx
   Updated 10 configuration examples to remove the nested "llm" key.
   Updated the configuration parameters table to match the code.
   Corrected the field name custom_prompt to scoring_prompt to match LLMRerankerConfig.

2. docs/components/rerankers/custom-prompts.mdx
   Updated the configuration example structure.
   Renamed custom_prompt to scoring_prompt.
   Renamed top_n to top_k to match the class definition.

3. docs/open-source/features/reranker-search.mdx
   Updated the reranker search configuration example to use the flat structure.

Fixes #3803  (issue)

## Type of change

- [x] Documentation update

## How Has This Been Tested?

I performed a static analysis and manual verification of the documentation against the codebase to ensure consistency.

- [x] Unit Test (Used as a general label for code verification)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/mem0ai/mem0?pullRequest=3808) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/mem0ai/mem0?pullRequest=3808) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/mem0ai/mem0?pullRequest=3808) it.</sub>

parthshr370 commented:
This LGTM @priyanshujaiz ! 
can you sign the CLA please and then we can start the review process

parthshr370 commented:
@priyanshujaiz , PR looks good , thanks for addressing this issue , can you also make sure that the models listed ( anthropic , OPENAI , etc ) are latest ones , or recommended ones , if you want to use the depricated models can you give the reasoning for using so

priyanshujaiz commented:
## Description

Fixed incorrect nested configuration structure in LLM reranker documentation and updated to use cost-effective recommended models.

Fixes #3803 (if there's an issue number)

## Changes Made

### 1. Configuration Structure Fix
- Removed nested `"llm"` structure that doesn't match code implementation
- Updated to flat config structure matching `LLMRerankerConfig` class
- Changed `custom_prompt` → `scoring_prompt` (correct field name)
- Changed `top_n` → `top_k` in custom-prompts.mdx

### 2. Model Updates
- Updated examples to use `gpt-4o-mini` (matches code default, 60x cheaper)
- Replaced deprecated `gpt-3.5-turbo` with `gpt-4o-mini`
- Updated Claude to `claude-3-5-sonnet-20241022` (latest version)
- Updated Azure OpenAI examples to use `gpt-4o-mini`
- Kept `gpt-4` only in quality config for comparison

### Files Updated:
1. `docs/components/rerankers/models/llm_reranker.mdx` (10 examples fixed + models updated)
2. `docs/components/rerankers/custom-prompts.mdx` (1 example fixed + model updated)
3. `docs/open-source/features/reranker-search.mdx` (1 example fixed + model updated)

## Type of change

- [x] Documentation update
- [x] Bug fix (fixes incorrect documentation)

## Verification

- [x] Cross-verified all config fields against `mem0/configs/rerankers/llm.py`
- [x] Verified structure matches `LLMReranker.__init__` implementation
- [x] Confirmed model recommendations align with code defaults
- [x] Branch synced with latest main (no conflicts)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] Documentation is now consistent with codebase implementation
